### PR TITLE
Fixes #31179 - Curls output during Global Registration

### PR DESCRIPTION
**Pull request by JaskaranNarula**
- https://projects.theforeman.org/issues/31179

### CODE REVIEW

**Review by ekohl — CHANGES_REQUESTED** [review]
I think this is wrong. This hides stderr, not stdout. stderr is important when debugging.

Looking at the original issue, I think `curl --silent` is a better alternative. Perhaps in combination with `--show-error`.

**Comment by JaskaranNarula** [comment]
Hello @ekohl, 

Thank you for reviewing the pr and suggesting the changes. I am fine with your suggestion as well. I tried to test and that also works like a charm. 

May be @stejskalleos also would have the same view on this. 

I wanted to ask one more thing, if this is the correct place for it. 

Do you suggest making a similar kind of change in the file mentioned below?
/usr/share/foreman/app/controllers/registration_controller.rb

**Comment by ekohl** [comment]
Perhaps @stejskalleos can answer that. I don't have a very good view on where that's used.

**Comment by stejskalleos** [comment]
@JaskaranNarula 
`--silent` looks like better alternative, all `curl` in `Global Registration Template` and `Linux Registration Default` should be called with `--silent` arg.

For the registration command (generated in `app/controllers/concerns/foreman/controller/registration_commands.rb`),
that could be also generated with `--silent` arg, but I would go for short version, like this:
`curl -sX GET ...`

File: app/views/unattended/provisioning_templates/registration/global_registration.erb
Comment (ekohl) [inline app/views/unattended/provisioning_templates/registration/global_registration.erb]: Small nit, but in scripts it's better to use the long form. In the controller it's intended to be copied so there it's ok. This script it too long to copy. Then it becomes more important to understand it when you read it. With `curl --silent` you instantly understand what's going on while with `curl -s` it's much harder unless you use curl very often. That's why I'd use `--silent` in the templates and `-s` in the controller.

File: app/controllers/registration_controller.rb
Comment (ekohl) [inline app/controllers/registration_controller.rb]: @stejskalleos `GET` is the default action so why is `-X GET` needed?

Comment (stejskalleos) [reply app/controllers/registration_controller.rb]: I think I wanted to distinct it from other curls in template that are to same `/register` path but with `POST`.
But I guess users don't care about it much, so it can be removed.

**Comment by ofedoren** [comment]
@stejskalleos, seems it needs a rebase.

**Comment by stejskalleos** [comment]
> stejskalleos, seems it needs a rebase.

I'm not the author, tagging @JaskaranNarula

**Comment by ofedoren** [comment]
@stejskalleos, my apologies :facepalm: Any Global Registration PRs I associate with you automatically :(

File: app/views/unattended/provisioning_templates/registration/global_registration.erb
Comment (ezr-ondrej) [inline app/views/unattended/provisioning_templates/registration/global_registration.erb]: Why are we still silencing the errors? :thinking:

**Review by ezr-ondrej — COMMENTED** [review]
Hi @JaskaranNarula, thanks for taking care of this!
This will need rebasing and also I thought the `-s` would be instead of silencing the `sterr`, why are we silencing even the errors?

File: app/views/unattended/provisioning_templates/registration/global_registration.erb
Comment (ekohl) [reply app/views/unattended/provisioning_templates/registration/global_registration.erb]: See https://github.com/theforeman/foreman/pull/8332#discussion_r580092605.

**Review by ekohl — CHANGES_REQUESTED** [review]
This is now waiting on @JaskaranNarula to address the comments.

**Comment by JaskaranNarula** [comment]
> This is now waiting on @JaskaranNarula to address the comments.

Thank you @ekohl @ezr-ondrej and @stejskalleos for reviewing the PR and for the suggestions. 
I will update the PR and will push the new changes and suggestions as mentioned in the above conversation.

**Comment by JaskaranNarula** [comment]
Hello @ekohl and @ezr-ondrej and @stejskalleos  
Please review the PR and then I will rebase it and then squash them into one.

File: app/views/unattended/provisioning_templates/registration/global_registration.erb
Comment (stejskalleos) [inline app/views/unattended/provisioning_templates/registration/global_registration.erb]: Can you remove the trailing space at the end of the row?

File: app/views/unattended/provisioning_templates/registration/global_registration.erb
Comment (stejskalleos) [inline app/views/unattended/provisioning_templates/registration/global_registration.erb]: Can you remove the trailing space at the end of the row?

**Review by stejskalleos — COMMENTED** [review]
Just one nit with trailing spaces,
otherwise looks good.

**Review by ekohl — COMMENTED** [review]
This looks good, but it appears there's a conflict now. Would you mind rebasing it on current develop to fix the merge conflict? Let us know if you need help with that.

**Comment by stejskalleos** [comment]
:+1: Changes LGTM,
can be merged.

**Comment by ekohl** [comment]
ok to test

**Review by ekohl — DISMISSED** [review]

**Comment by ekohl** [comment]
It looks like the tests need to be modified to reflect these changes. I'm not sure what's up with Katello but I think that's unrelated.

**Comment by ezr-ondrej** [comment]
Yeah, the tests needs to be fixed, which should be quite straight forward:
https://github.com/theforeman/foreman/blob/245d23de57e306ee1efec18ee69152f176ed3e17/test/controllers/api/v2/registration_commands_controller_test.rb#L10
https://github.com/theforeman/foreman/blob/245d23de57e306ee1efec18ee69152f176ed3e17/test/controllers/api/v2/registration_commands_controller_test.rb#L47

**Comment by JaskaranNarula** [comment]
@ezr-ondrej, 

Thanks for pointing out the fix in the test. 

I am working on fixing it. Just want to know do we need to put --silent or -s will work on both the lines?

**Comment by ezr-ondrej** [comment]
> Just want to know do we need to put --silent or -s will work on both the lines?

Both are testing the controller and there is `-s` so we should expect -s in both the tests.

BTW you can run the test locally by `bundle exec rake test TEST=test/controllers/api/v2/registration_commands_controller_test.rb`

**Comment by ezr-ondrej** [comment]
Thanks for the separate commits, it was easier to review and it works :+1:
But I've squashed them for you, because the second one had invalid commit message, so the `Redmine issues` check didn't pass and I can't merge without it passing :)

**Review by ezr-ondrej — APPROVED** [review]
Thanks @JaskaranNarula :+1:

**Comment by JaskaranNarula** [comment]
Thank you @ezr-ondrej, 

> Thanks for the separate commits, it was easier to review and it works 
> But I've squashed them for you, because the second one had invalid commit message, so the `Redmine issues` check didn't pass and I can't merge without it passing :)

To be honest, I was waiting for the test to complete with success and then I thought that for the commit message I will do the append. But Thanks to you :) :D

**Comment by ezr-ondrej** [comment]
> test to complete with success

I was waiting for the same with the squash xD so I was just a little bit faster. Katello will probably fail again, but if related, we should just open PR before merging, if unrelated (most likely) I'll just merge ;)

**Pull request merged**
